docs/en/guides/pipeline-integrations/development/sgtk-developer-app.md

```diff
@@ -0,0 +1,257 @@
+The workflow is now identical to the one which happens with official app store apps.
+
+{% include warning title="Caution" content="The git descriptor works well with [centralized configs](https://developer.shotgunsoftware.com/tk-core/initializing.html#centralized-configurations), where the caching of apps is usually run by an admin and is stored to a central location where all user's can access it. However, if you are using a [distributed config](https://developer.shotgunsoftware.com/tk-core/initializing.html#distributed-configurations), then it may not be as suitable. Your app will be downloaded per user, which means each user will need to have git installed and setup to authenticate with your repo, and access the code." %}
+
+## Modifying an existing app
+Rather than starting from an empty starter template, it is sometimes necessary to add a minor feature to an existing app, for example, one of Shotgun Software's standard apps. 
+When you work with a modified version of an app, you typically want to 'track' against the source app and make sure to regularly pull in changes and bug fixes.
+
+When you do this type of development, you pick up the parent code, then apply some of your changes, then release this to your pipeline. 
+Your release effectively consists of the base version of the app PLUS your applied local changes. 
+We recommend that you add a version suffix to the existing version number. 
+This will work seamlessly with Toolkit and is relatively straight forward. 
+The following workflow illustrates how to proceed:
+
+- You fork the parent app and create your own repository. With the fork, you get all the git tags. 
+Let's say the latest tag is called `v0.2.12` and the master branch basically equals the contents in this tag.
+- You apply your changes and commit to your master branch. You now have `v0.2.12` PLUS your changes. 
+When you release this to production you need to create a tag. Name the tag `v0.2.12.1`, to indicate that your code is based on `v0.2.12` and it is the first release.
+- Now someone finds a bug in your modifications. Fix the bug and tag up and release `v0.2.12.2`.
+- A number of important bug fixes have been released in the parent repository. 
+Pull them down to your repository. The most recent tag is now `v0.2.15` due to releases that have happened in the parent repository. 
+Now merge your changes with master and test. You now basically have parent app `v0.2.15` merged with your changes. Tag up `v0.2.15.1`.
+
+The tagging scheme outlined above guarantees that the Toolkit updates work correctly and make it easy to quickly see what code each tag in your fork is based on.
+
+### Pull requests
+We love pull requests! If you feel you have made a change that can benefit others, don't hesitate to feed it back to us as a pull request.
+We may then fold it back into the main version of the app.
+Alternatively, add a suggestion for a new idea on our [roadmap page](https://www.shotgunsoftware.com/roadmap/).
```

Comment:
We mention Community and the Community Shared Integrations page in the engine doc, but do we want to mention them here as well? good to plug the forums, and good for people to know that they have a place to have conversations around this stuff, and good to suggest to them not just to submit PRs, but that they can share an entire app with the community if they feel moved to do so.

Reply:
That didn't really fit under the head of Pull Requests, so I've renamed the heading to "Contributing".
and added:

> If you've made an entire app that your willing to share with the community, let us all know on the [forums](https://community.shotgunsoftware.com/)!

Reply:
perfect!